internal/logs/workload.go

```diff
@@ -53,11 +54,12 @@ func (t *transportLog) Stop() {
 }
 
 type WorkloadsLogsTarget struct {
-	deviceID        string
-	transports      map[string]transportLog
-	wwriter         map[string]*WorkloadWriter // key is the workload name
-	workloadsConfig map[string]string
-	workloadManager *workload.WorkloadManager
+	deviceID            string
+	transports          map[string]transportLog
+	wwriter             map[string]*WorkloadWriter // key is the workload name
+	workloadsConfig     map[string]string
+	workloadManager     *workload.WorkloadManager
+	workloadsConfigLock sync.Mutex
```

Comment:
Based on the use of this lock, will [RWMutex](https://pkg.go.dev/sync#RWMutex) be a better fit where it only locks for write in `Update` function, but allows multiple read in `WorkloadStarted` function?

Reply:
I was not aware of such Mutex but generally speaking, yes, it would be better to use that kind of mutex here or even everywhere where there are more read locked than writes (see other PR with data races)

Reply:
Please use the RWMutex instead.

Reply:
@jakub-dzon pls re-review